bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVGsonDTO.java

```diff
@@ -0,0 +1,194 @@
+     */
+    public static class MRPayEvent {
+        @SerializedName("new_play_mode")
+        public Integer newPlayMode = EV_PLAYCHG_STOP;
+        public Integer duration = -1;
+        public Integer playBackState = EV_PLAYCHG_STOP;
+        public Integer mediaType = 0;
+        public String mediaCode = "";
+        public Integer playPostion = -1;
+    }
+
+    public static class MRPayEventInstanceCreator implements InstanceCreator<MRPayEvent> {
+        @Override
+        public MRPayEvent createInstance(@Nullable Type type) {
+            return new MRPayEvent();
+        }
+    }
+
+    /**
+     * Deutsche Telekom uses a OAuth-based authentication to access the EPG portal.
+     * The binding automates the login incl. OAuth authentication. This class helps mapping the response to a Java
+     * object (using Gson)
+     *
+     * Sample response:
+     * { "enctytoken":"7FA9A6C05EDD873799392BBDDC5B7F34","encryptiontype":"0002",
+     * "platformcode":"0200", "epgurl":"http://appepmfk20005.prod.sngtv.t-online.de:33200",
+     * "version":"MEM V200R008C15B070", "epghttpsurl":"https://appepmfk20005.prod.sngtv.t-online.de:33207",
+     * "rootCerAddr": "http://appepmfk20005.prod.sngtv.t-online.de:33200/EPG/CA/iptv_ca.der",
+     * "upgAddr4IPTV":"https://slbedifk11100.prod.sngtv.t-online.de:33428/EDS/jsp/upgrade.jsp",
+     * "upgAddr4OTT":"https://slbedmfk11100.prod.sngtv.t-online.de:33428/EDS/jsp/upgrade.jsp,https://slbedmfk11100.prod.sngtv.t-online.de:33428/EDS/jsp/upgrade.jsp",
+     * "sam3Para": [
+     * {"key":"SAM3ServiceURL","value":"https://accounts.login.idm.telekom.com"},
+     * {"key":"OAuthClientSecret","value":"21EAB062-C4EE-489C-BC80-6A65397F3F96"},
+     * {"key":"OAuthScope","value":"ngtvepg"},
+     * {"key":"OAuthClientId","value":"10LIVESAM30000004901NGTV0000000000000000"} ]
+     * }
+     */
+    public static class OauthCredentials {
+        public String epghttpsurl = "";
+        public ArrayList<OauthKeyValue> sam3Para = new ArrayList<OauthKeyValue>();
+    }
+
+    public static class OauthCredentialsInstanceCreator implements InstanceCreator<OauthCredentials> {
+        @Override
+        public OauthCredentials createInstance(@Nullable Type type) {
+            return new OauthCredentials();
+        }
+    }
+
+    public static class OauthKeyValue {
+        public String key = "";
+        public String value = "";
+    }
+
+    public static class OAuthTokenResponse {
+        @SerializedName("error_description")
+        public String errorDescription = "";
+        public String error = "";
+        @SerializedName("access_token")
+        public String accessToken = "";
+    }
+
+    public static class OAuthTokenResponseInstanceCreator implements InstanceCreator<OAuthTokenResponse> {
+        @Override
+        public OAuthTokenResponse createInstance(@Nullable Type type) {
+            return new OAuthTokenResponse();
+        }
+    }
+
+    public static class OAuthAutenhicateResponse {
```

Comment:
Typo, should be `OAuthAuthenticateResponse`

Reply:
fixed